### DISCUSSION TITLE
如何取得認證

### DISCUSSION THREAD

**Discussion by Redpanda1235** [osk2/panasonic_smart_app#52] [Q&A]
Hi,
你好，我剛購入禾聯一個新的智慧家電，
想自己整合到HA裡，
我了解需要登入自己的帳號才可以控制，
請問要怎麼過家電廠的登入認證呢？

把Androird APK 反編譯 看看裡面的程式碼嗎？
還是有更好的方法？

**Comment by osk2** [comment]
這有蠻多方法的，然後各家實作方式不同也沒有一個標準答案
我自己是在虛擬機安裝 apk，攔截虛擬機流量，過濾出 apk 發出的請求，再自己整理每隻 API 的請求和回應

建議你到 [Taiwan Home Assistant 同好交流](https://www.facebook.com/groups/151166072456061) 尋求幫助會較有收穫
